checker/hostname.go

```diff
@@ -210,29 +210,40 @@ func checkTLSVersion(client *smtp.Client, hostname string, timeout time.Duration
 	return result.Success()
 }
 
-// CheckHostnameWithCache returns the result of CheckHostname, using or
-// updating the Checker's cache.
-func (c *Checker) CheckHostnameWithCache(domain string, hostname string) HostnameResult {
+// checkHostname returns the result of c.CheckHostname or FullCheckHostname,
+// using or updating the Checker's cache.
+func (c *Checker) checkHostname(domain string, hostname string) HostnameResult {
+	check := c.CheckHostname
+	if check == nil {
+		// If CheckHostname hasn't been set, default to the full set of checks.
+		check = FullCheckHostname
+	}
+
 	if c.Cache == nil {
-		return c.CheckHostname(domain, hostname)
+		return check(domain, hostname, c.timeout())
 	}
 	hostnameResult, err := c.Cache.GetHostnameScan(hostname)
 	if err != nil {
-		hostnameResult = c.CheckHostname(domain, hostname)
+		hostnameResult = check(domain, hostname, c.timeout())
 		c.Cache.PutHostnameScan(hostname, hostnameResult)
 	}
 	return hostnameResult
 }
 
-// CheckHostname performs a series of checks against a hostname for an email domain.
-// `domain` is the mail domain that this server serves email for.
-// `hostname` is the hostname for this server.
-func (c *Checker) CheckHostname(domain string, hostname string) HostnameResult {
-	if c.checkHostnameOverride != nil {
-		// Allow the Checker to mock this function.
-		return c.checkHostnameOverride(domain, hostname)
+func NoopCheckHostname(domain string, hostname string, _ time.Duration) HostnameResult {
```

Comment:
```suggestion
// NoopCheckHostname returns a fake error result containing `domain` and `hostname`.
func NoopCheckHostname(domain string, hostname string, _ time.Duration) HostnameResult {
```
oor something like that? to appease the linter

Reply:
Yess thank you for linting my code :)